# Comparing `tmp/genomenotebook-0.2.tar.gz` & `tmp/genomenotebook-0.2.1.tar.gz`

## filetype from file(1)

```diff
@@ -1 +1 @@
-gzip compressed data, was "genomenotebook-0.2.tar", max compression
+gzip compressed data, was "genomenotebook-0.2.1.tar", last modified: Mon Apr 17 18:57:59 2023, max compression
```

## Comparing `genomenotebook-0.2.tar` & `genomenotebook-0.2.1.tar`

### file list

```diff
@@ -1,5 +1,23 @@
--rw-r--r--   0        0        0      785 2023-03-27 17:02:10.386248 genomenotebook-0.2/README.md
--rw-r--r--   0        0        0       36 2023-03-27 17:02:10.434247 genomenotebook-0.2/genomenotebook/__init__.py
--rw-r--r--   0        0        0     8490 2023-03-27 17:02:10.434247 genomenotebook-0.2/genomenotebook/genomenotebook.py
--rw-r--r--   0        0        0     1466 2023-03-27 17:02:10.434247 genomenotebook-0.2/pyproject.toml
--rw-r--r--   0        0        0     1889 1970-01-01 00:00:00.000000 genomenotebook-0.2/PKG-INFO
+drwxr-xr-x   0 dbikard   (1000) dbikard   (1000)        0 2023-04-17 18:57:59.888870 genomenotebook-0.2.1/
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)    11357 2023-04-17 08:46:32.000000 genomenotebook-0.2.1/LICENSE
+-rw-rw-r--   0 dbikard   (1000) dbikard   (1000)      111 2023-01-20 02:50:04.000000 genomenotebook-0.2.1/MANIFEST.in
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)     1219 2023-04-17 18:57:59.888870 genomenotebook-0.2.1/PKG-INFO
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)      536 2023-04-17 15:11:04.000000 genomenotebook-0.2.1/README.md
+drwxr-xr-x   0 dbikard   (1000) dbikard   (1000)        0 2023-04-17 18:57:59.888870 genomenotebook-0.2.1/genomenotebook/
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)       22 2023-04-17 14:38:45.000000 genomenotebook-0.2.1/genomenotebook/__init__.py
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)     3722 2023-04-17 14:38:45.000000 genomenotebook-0.2.1/genomenotebook/_modidx.py
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)     7357 2023-04-17 14:38:45.000000 genomenotebook-0.2.1/genomenotebook/browser.py
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)      294 2023-04-17 14:38:45.000000 genomenotebook-0.2.1/genomenotebook/examples.py
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)      642 2023-04-14 08:24:05.000000 genomenotebook-0.2.1/genomenotebook/js_callback_code.py
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)     7515 2023-04-17 14:38:45.000000 genomenotebook-0.2.1/genomenotebook/plotting.py
+drwxr-xr-x   0 dbikard   (1000) dbikard   (1000)        0 2023-04-17 18:57:59.888870 genomenotebook-0.2.1/genomenotebook.egg-info/
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)     1219 2023-04-17 18:57:59.000000 genomenotebook-0.2.1/genomenotebook.egg-info/PKG-INFO
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)      486 2023-04-17 18:57:59.000000 genomenotebook-0.2.1/genomenotebook.egg-info/SOURCES.txt
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)        1 2023-04-17 18:57:59.000000 genomenotebook-0.2.1/genomenotebook.egg-info/dependency_links.txt
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)       50 2023-04-17 18:57:59.000000 genomenotebook-0.2.1/genomenotebook.egg-info/entry_points.txt
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)        1 2023-04-17 09:24:03.000000 genomenotebook-0.2.1/genomenotebook.egg-info/not-zip-safe
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)       88 2023-04-17 18:57:59.000000 genomenotebook-0.2.1/genomenotebook.egg-info/requires.txt
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)       15 2023-04-17 18:57:59.000000 genomenotebook-0.2.1/genomenotebook.egg-info/top_level.txt
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)     1027 2023-04-17 14:28:03.000000 genomenotebook-0.2.1/settings.ini
+-rw-r--r--   0 dbikard   (1000) dbikard   (1000)       38 2023-04-17 18:57:59.888870 genomenotebook-0.2.1/setup.cfg
+-rw-rw-r--   0 dbikard   (1000) dbikard   (1000)     2560 2023-01-20 02:50:04.000000 genomenotebook-0.2.1/setup.py
```

### Comparing `genomenotebook-0.2/genomenotebook/genomenotebook.py` & `genomenotebook-0.2.1/genomenotebook/plotting.py`

 * *Files 18% similar despite different names*

```diff
@@ -1,82 +1,28 @@
+# AUTOGENERATED! DO NOT EDIT! File to edit: ../nbs/01_plotting.ipynb.
+
+# %% auto 0
+__all__ = ['Y_RANGE', 'gene_y_range', 'create_genome_browser_plot', 'get_gene_patches', 'get_y_range', 'get_genome_annotations',
+           'get_genes_from_annotation', 'get_all_glyphs', 'rect_patch', 'arrow_patch', 'get_arrow_patch']
+
+# %% ../nbs/01_plotting.ipynb 4
+import gffpandas.gffpandas as gffpd
 import numpy as np
-from Bio import SeqIO
 
 from bokeh.plotting import figure
 from bokeh.models.tools import BoxZoomTool
 from bokeh.models import HoverTool, NumeralTickFormatter, LabelSet
 from bokeh.models.glyphs import Patches
-
-import gffpandas.gffpandas as gffpd
 from bokeh.models import (
     CustomJS,
     Range1d,
     ColumnDataSource,
 )
 
-from bokeh.io import output_notebook
-output_notebook()
-
-init_pos = 10000
-init_win = 20000
-
-
-def browser(genome_path: str, gff_path: str, 
-             genome_format="fasta", 
-             init_pos=init_pos, 
-             init_win=init_win, 
-             **kwargs):
-    
-    genome=next(SeqIO.parse(genome_path,genome_format))
-    annotation = get_genome_annotations(gff_path)
-    annotation.gene=annotation.gene.fillna("")
-    annotation.gene=annotation[['gene','locus_tag']].apply(lambda x: f"{x[0]} {x[1]}",axis=1) #create names with both locus tag and gene name
-    genes = get_genes_from_annotation(annotation)
-
-    semi_win = init_win / 2
-    init_left = init_pos - semi_win
-    init_right = init_pos + semi_win
-    x_range = Range1d(
-            init_left, init_right, bounds=(init_left, init_right), 
-            max_interval = 100000,
-            min_interval = 50,
-        )
-    glyphSource = ColumnDataSource(get_gene_patches(genes, init_left, init_right))
-    p=create_genome_browser_plot(glyphSource,x_range,**kwargs)
-
-
-    sequence = {'base': list(str(genome.seq[int(init_left):int(init_right)]).upper()),
-                                'x': list(range(int(init_left),int(init_right)))}
-
-    seqSource = ColumnDataSource({'base': [" "],
-                                'x': [0]})
-    seqLabel = LabelSet(x='x', y=-1.98, text='base',
-                      text_font_size='8pt', 
-                      text_font_style='bold',
-                      source=seqSource)
-    p.add_layout(seqLabel)
-
-    xcb= CustomJS(args={"x_range":p.x_range,"seqSource":seqSource,"sequence":sequence}, code="""
-        let x_size = x_range.end - x_range.start;
-        const seqdata = seqSource.data;
-        const x = seqSource.data['x'];
-        if (x_size<300){
-            seqSource.data['base']=sequence['base'].slice(x_range.start-sequence['x'][0], x_range.end-sequence['x'][0]);
-            seqSource.data['x']=sequence['x'].slice(x_range.start-sequence['x'][0], x_range.end-sequence['x'][0]);
-        } else {
-            seqSource.data['base']=[" "];
-            seqSource.data['x']=[0]
-        }
-        seqSource.change.emit()
-    """)
-
-    p.x_range.js_on_change('start', xcb)
-    p.x_range.js_on_change('end', xcb)
-    return p
-
+# %% ../nbs/01_plotting.ipynb 5
 def create_genome_browser_plot(glyphSource, x_range, **kwargs):
     plot_height = kwargs.get("plot_height", 150)
     label_angle = kwargs.get("label_angle", 45)
     text_font_size = kwargs.get("text_font_size", "10pt")
     output_backend = kwargs.get("output_backend", "webgl")
     
     y_min, y_max = get_y_range()
@@ -124,96 +70,15 @@
         HoverTool(
             renderers=[glyph],
             tooltips=[("name", "@hover_names"), ("product", "@product")],
         )
     )
     return p_annot
 
-
-
-gene_y_range = (-1.5, -1)
-
-def get_arrow_patch(genes_region, ori="+"):
-    y_min, y_max = gene_y_range
-    y_min = y_min 
-    if ori == "+":
-        xs = list(
-            zip(
-                genes_region.start.values,
-                genes_region.start.values,
-                np.maximum(genes_region.start.values, genes_region.end.values - 100),
-                genes_region.end.values,
-                np.maximum(genes_region.start.values, genes_region.end.values - 100),
-            )
-        )
-        color = ["orange"] * genes_region.shape[0]
-    elif ori == "-":
-        xs = list(
-            zip(
-                genes_region.start.values,
-                genes_region.start.values,
-                np.minimum(genes_region.start.values, genes_region.end.values + 100),
-                genes_region.end.values,
-                np.minimum(genes_region.start.values, genes_region.end.values + 100),
-            )
-        )
-        color = ["purple"] * genes_region.shape[0]
-
-    ys = [
-        np.array([y_min, y_max, y_max, (y_max + y_min) / 2, y_min])
-        for i in range(genes_region.shape[0])
-    ]
-    genes_mid = genes_region.left + (genes_region.right - genes_region.left) / 2
-    pos = list(genes_mid.values)
-    names = list(genes_region.gene.values)
-    product = list(genes_region["product"].values)
-    return dict(
-        xs=xs,
-        ys=ys,
-        pos=pos,
-        names=names,
-        hover_names=names,
-        product=product,
-        color=color,
-    )
-
-def arrow_patch(genes_region):
-    arr_plus = get_arrow_patch(genes_region[genes_region["strand"] == "+"], "+")
-    arr_minus = get_arrow_patch(genes_region[genes_region["strand"] == "-"], "-")
-    return dict([(k, arr_plus[k] + arr_minus[k]) for k in arr_plus.keys()])
-
-
-def rect_patch(genes_region):
-    y_min, y_max = gene_y_range
-    xs = list(
-        zip(
-            genes_region.start.values,
-            genes_region.start.values,
-            genes_region.end.values,
-            genes_region.end.values,
-        )
-    )
-    xs = [np.array(x) for x in xs]
-    ys = [np.array([y_min, y_max, y_max, y_min]) for i in range(genes_region.shape[0])]
-    genes_mid = genes_region.left + (genes_region.right - genes_region.left) / 2
-    pos = list(genes_mid.values)
-    names = list(genes_region.gene.values)
-    product = list(genes_region["product"].values)
-    color = ["grey"] * genes_region.shape[0]
-    return dict(
-        xs=xs,
-        ys=ys,
-        pos=pos,
-        names=[""] * genes_region.shape[0],
-        hover_names=names,
-        product=product,
-        color=color,
-    )
-
-
+# %% ../nbs/01_plotting.ipynb 6
 def get_gene_patches(genes, left, right):
     genes_region = genes[
         (genes["right"] > left)
         & (genes["left"] < right)
         & (genes["type"] != "repeat_region")
     ]
     arr = arrow_patch(genes_region)
@@ -236,17 +101,20 @@
 
     :return: [description]
     :rtype: [type]
     """
     return Y_RANGE
 
 
-def get_genome_annotations(genome_path: str):
+def get_genome_annotations(genome_path: str, bounds=None):
     annotation = gffpd.read_gff3(genome_path)
     annotation = annotation.attributes_to_columns()
+    if bounds:
+        annotation = annotation.loc[(annotation.start<bounds[1]) & (annotation.end>bounds[0])]
+
     annotation.loc[:, "left"] = annotation[["start"]].values
     annotation.loc[:, "right"] = annotation[["end"]].values
     return annotation
 
 def get_genes_from_annotation(annotation):
 
     genes = annotation[
@@ -266,8 +134,105 @@
     ].values
 
     genes.loc[genes["strand"] == "-", "end"] = genes.loc[
         genes["strand"] == "-", "left"
     ].values
 
     genes.loc[genes["type"] == "repeat_region", "gene"] = "REP"
-    return genes
+    return genes
+
+def get_all_glyphs(genes,bounds:tuple):
+    all_glyphs=get_gene_patches(genes, bounds[0], bounds[1])
+
+    ks=list(all_glyphs.keys())
+    ref_list_ix=ks.index('xs')
+    # Sort all the lists in the dictionary based on the values of the reference list
+    sorted_lists = sorted(zip(*[all_glyphs[k] for k in ks]), key= lambda x: x[ref_list_ix][0])
+
+    # Convert the sorted tuples back into separate lists
+    unzipped_lists = zip(*sorted_lists)
+
+    # Create a new dictionary with the same keys as the original dictionary, but with the sorted lists as values
+    all_glyphs = {k: list(t) for k, t in zip(ks, unzipped_lists)}
+    
+    return all_glyphs
+
+# %% ../nbs/01_plotting.ipynb 7
+def rect_patch(genes_region):
+    y_min, y_max = gene_y_range
+    xs = list(
+        zip(
+            genes_region.start.values,
+            genes_region.start.values,
+            genes_region.end.values,
+            genes_region.end.values,
+        )
+    )
+    xs = [np.array(x) for x in xs]
+    ys = [np.array([y_min, y_max, y_max, y_min]) for i in range(genes_region.shape[0])]
+    genes_mid = genes_region.left + (genes_region.right - genes_region.left) / 2
+    pos = list(genes_mid.values)
+    names = list(genes_region.gene.values)
+    product = list(genes_region["product"].values)
+    color = ["grey"] * genes_region.shape[0]
+    return dict(
+        xs=xs,
+        ys=ys,
+        pos=pos,
+        names=[""] * genes_region.shape[0],
+        hover_names=names,
+        product=product,
+        color=color,
+    )
+
+# %% ../nbs/01_plotting.ipynb 8
+def arrow_patch(genes_region):
+    arr_plus = get_arrow_patch(genes_region[genes_region["strand"] == "+"], "+")
+    arr_minus = get_arrow_patch(genes_region[genes_region["strand"] == "-"], "-")
+    return dict([(k, arr_plus[k] + arr_minus[k]) for k in arr_plus.keys()])
+
+# %% ../nbs/01_plotting.ipynb 9
+gene_y_range = (-1.5, -1)
+
+def get_arrow_patch(genes_region, ori="+"):
+    y_min, y_max = gene_y_range
+    y_min = y_min 
+    if ori == "+":
+        xs = list(
+            zip(
+                genes_region.start.values,
+                genes_region.start.values,
+                np.maximum(genes_region.start.values, genes_region.end.values - 100),
+                genes_region.end.values,
+                np.maximum(genes_region.start.values, genes_region.end.values - 100),
+            )
+        )
+        color = ["orange"] * genes_region.shape[0]
+    elif ori == "-":
+        xs = list(
+            zip(
+                genes_region.start.values,
+                genes_region.start.values,
+                np.minimum(genes_region.start.values, genes_region.end.values + 100),
+                genes_region.end.values,
+                np.minimum(genes_region.start.values, genes_region.end.values + 100),
+            )
+        )
+        color = ["purple"] * genes_region.shape[0]
+
+    ys = [
+        np.array([y_min, y_max, y_max, (y_max + y_min) / 2, y_min])
+        for i in range(genes_region.shape[0])
+    ]
+    genes_mid = genes_region.left + (genes_region.right - genes_region.left) / 2
+    pos = list(genes_mid.values)
+    names = list(genes_region.gene.values)
+    product = list(genes_region["product"].values)
+    return dict(
+        xs=xs,
+        ys=ys,
+        pos=pos,
+        names=names,
+        hover_names=names,
+        product=product,
+        color=color,
+    )
```

